sycl/test-e2e/ESIMD/dword_atomic_accessor_smoke_stateless.cpp

```diff
@@ -0,0 +1,24 @@
+//==-dword_atomic_accessor_smoke_stateless.cpp - DPC++ ESIMD on-device test-==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// This test checks DWORD atomic operations when stateless memory accesses are
+// enforced, i.e. accessor based accesses are automatically converted to
+// stateless accesses.
+//===----------------------------------------------------------------------===//
+// REQUIRES: gpu
+// UNSUPPORTED: gpu-intel-gen9 && windows
+// UNSUPPORTED: cuda || hip
+// RUN: %clangxx -fsycl -fsycl-esimd-force-stateless-mem %s -o %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+// TODO add support for __esimd_dword_atomic on esimd_emulator
+// XFAIL: esimd_emulator
```

Comment:
if we are sure it always fails these is fine, if it could ever pass we should use `UNSUPPORTED`